### DISCUSSION TITLE
Debugging nanoVNASaver  - seeking an alterrantive to Eclipse

### DISCUSSION THREAD

**Discussion by anneranch** [NanoVNA-Saver/nanovna-saver#649] [Q&A]
I cannot reliably duplicate nanoVNASaver code using Eclipse. 
There are THREE options to "clone" and NONE of them 
repeatably builds PyDev project. 
Since the code is written in PyQt6 - what would be an alternative IDE to "reverse engineer" 
nanoVNASaver?  It looks as PyQt6 is NOT IDE, but I may  be wrong. 
I am NOT ready to publish my code modifications -
this is strictly a private effort. 
Not to worry.

**Comment by mihtjel** [comment]
The original version was made using PyCharm from JetBrains - which is available in a free community edition.

Don't forget that the project is licensed under the GPL, so that any modifications that are distributed must also make the source code available to the recipients. Your own personal use is of course fully up to you :-)

**Reply by anneranch** [reply]
Thanks,  I did check PcCharm and having an issue with cloning.  It asks for 
authorization before I can actually clone , while using Eclipse "git" wizards  there is  no need for authorization. 
I have finally installed Eclipse 2023-06 and see some new "stepping  thru the Python code" options.
